pymepixviewer/dialogs/postprocessing.py

```diff
@@ -0,0 +1,112 @@
+        self.lineEditTimeWalkFile.setText(file_name)
+
+    def onBrowseTimeWalkFileCentroided(self):
+        file_name, _ = QtGui.QFileDialog.getOpenFileName(self, 'Choose centroided time walk file', None, 'All files (*.*);;Numpy (*.npy)', 'Numpy (*.npy)')
+        self.lineEditTimeWalkFileCentroided.setText(file_name)
+
+    def onBrowseOutputDirectory(self):
+        output_directory = QtGui.QFileDialog.getExistingDirectory(self, 'Choose output directory')
+        self.lineEditOutputDirectory.setText(output_directory)
+
+    def updateProgressBar(self, progress):
+        progress = int(((self.files_completed / self.files_for_processing) + progress / self.files_for_processing) * 100)
+        self.progressBar.setValue(progress)
+
+    def onStartProcessing(self):
+        if not self.processing_is_running:
+            self.processing_thread = Thread(target = self.run_post_processing_threaded)
+            self.processing_thread.start()
+        else:
+            self.pushButtonStartProcessing.setDisabled(True)
+            self.pushButtonStartProcessing.setText('Stopping')
+            self.setToolTip("Stopping after currently processed file. This can still take a while!")
+            self.processing_is_running = False
+
+    def __set_disabled_controls(self, disabled):
+        self.pushButtonBrowseInputFiles.setDisabled(disabled)
+        self.pushButtonBrowseOutputDirectory.setDisabled(disabled)
+        self.pushButtonBrowseTimeWalkFile.setDisabled(disabled)
+        self.pushButtonBrowseTimeWalkFileCentroided.setDisabled(disabled)
+
+        self.lineEditInputFiles.setDisabled(disabled)
+        self.lineEditOutputDirectory.setDisabled(disabled)
+        self.lineEditTimeWalkFile.setDisabled(disabled)
+        self.lineEditTimeWalkFileCentroided.setDisabled(disabled)
+    
+
+    def run_post_processing_threaded(self):
+        self._close_allowed = False
+        self.pushButtonStartProcessing.setText('Stop Processing')
+        self.setToolTip("Stop after currently processed file. This can still take a while!")
+        self.__set_disabled_controls(True)
+        self.processing_is_running = True
+
+        input_files = self.lineEditInputFiles.text().split(FILE_SEPARATOR)
+        self.files_for_processing = len(input_files)
+        self.files_completed = 0
+        time_walk_file = self.lineEditTimeWalkFile.text()
+        if len(time_walk_file) == 0:
+            time_walk_file = None
+        time_walk_file_centroided = self.lineEditTimeWalkFileCentroided.text()
+        if len(time_walk_file_centroided) == 0:
+            time_walk_file_centroided = None
+
+        output_directory = self.lineEditOutputDirectory.text()
+        
+        for input_file in input_files:
+            self.progressBar.setToolTip(f'Processing: {input_file}')
+            if not self.processing_is_running:
+                break
+            output_file = f'{output_directory}/{Path(input_file).stem}.{OUTPUT_FILE_EXTENSION}'
```

Comment:
I think, I already commented this in the previous review. Use `pathlib` or `os.path`.